### PR TITLE
add Adafruit metro S3 16MB Flash, 8MB PSRam

### PR DESCRIPTION
**Pull request by hathach**
## Description of Change

- Add Adafruit metro S3 16MB Flash, 8MB PSRam https://www.adafruit.com/product/5500
- also reorder to have feather boards next to each others

## Tests scenarios

Tested with S3 board with latest arduino-esp32 (2.0.11)

@ladyada

### CODE REVIEW

**Comment by ladyada** [comment]
verifed on proto hardware - thanks ! :)

**Comment by kattni** [comment]
@VojtechBartoska Hello! We are close to releasing this board. I wanted to check in with you about getting it added. Please let us know if there's anything else we need to do. Thank you!

**Comment by me-no-dev** [comment]
this will be merged prior to releasing 2.0.12 (probably next week or so)

**Comment by VojtechBartoska** [comment]
added to 2.0.12, @ladyada 2.0.12 is supposed to be last bug fix release based on IDF 4.4. We'll after that focus just on 3.0 (IDF 5.1), please consider this for adding new boards support.

**Comment by ladyada** [comment]
sounds great - we're also moving to 5.1 for circuitpython :)
having fun with the RGB TFT driving ability of the S3 
image

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by hathach** [comment]
hihi, we want to make some last minute changes (minor) for this board. Please hold on a bit before merging. thank you :)

**Comment by P-R-O-C-H-Y** [comment]
@hathach ping me when its ready please :)

**Comment by hathach** [comment]
> @hathach ping me when its ready please :)

@P-R-O-C-H-Y thank you for waiting, just pushed the latest update, this is ready to merge now.

File: boards.txt [hunk 1]
```diff
@@ -8397,6 +8270,194 @@ adafruit_metro_esp32s2.menu.EraseFlash.none.upload.erase_cmd=
...
+adafruit_metro_esp32s3.build.flash_size=16MB
+adafruit_metro_esp32s3.build.flash_freq=80m
+adafruit_metro_esp32s3.build.flash_mode=dio
+adafruit_metro_esp32s3.build.boot=qio
+adafruit_metro_esp32s3.build.partitions=default
+adafruit_metro_esp32s3.build.defines=
+adafruit_metro_esp32s3.build.loop_core=
+adafruit_metro_esp32s3.build.event_core=
+adafruit_metro_esp32s3.build.flash_type=qio
+adafruit_metro_esp32s3.build.psram_type=qspi
```
Comment (P-R-O-C-H-Y) [inline boards.txt:8323]: I'm not now 100% sure, but OCTAL PSRAM cannot run in QUAD mode. You should as default set it:
`adafruit_metro_esp32s3.build.psram_type=opi`
`adafruit_metro_esp32s3.build.defines=-DBOARD_HAS_PSRAM`

File: boards.txt [hunk 2]
```diff
@@ -8397,6 +8270,194 @@ adafruit_metro_esp32s2.menu.EraseFlash.none.upload.erase_cmd=
...
+adafruit_metro_esp32s3.menu.UploadMode.default=UART0 / Hardware CDC
+adafruit_metro_esp32s3.menu.UploadMode.default.upload.use_1200bps_touch=false
+adafruit_metro_esp32s3.menu.UploadMode.default.upload.wait_for_upload_port=false
+
+adafruit_metro_esp32s3.menu.PSRAM.opi=OPI PSRAM
+adafruit_metro_esp32s3.menu.PSRAM.opi.build.defines=-DBOARD_HAS_PSRAM
+adafruit_metro_esp32s3.menu.PSRAM.opi.build.psram_type=opi
+adafruit_metro_esp32s3.menu.PSRAM.enabled=QSPI PSRAM
+adafruit_metro_esp32s3.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM
+adafruit_metro_esp32s3.menu.PSRAM.enabled.build.psram_type=qspi
```
Comment (P-R-O-C-H-Y) [inline boards.txt:8368]: If my previous comment is true, this part should be removed :)

**Review by P-R-O-C-H-Y — COMMENTED** [review]
@hathach please take a look on my comment :)

File: boards.txt (hunk 2 above)
Comment (P-R-O-C-H-Y) [reply boards.txt:8368]: You can test it easily on your new board, but I guess the board will just crash/reboots infinitely.

Comment (P-R-O-C-H-Y) [reply boards.txt:8368]: @hathach So its confirmed now, would be better for users to remove this, if the board will sell only with OPI PSRAM.

Comment (hathach) [reply boards.txt:8368]: thank you, Will follow your suggestion

**Comment by hathach** [comment]
> @hathach please take a look on my comment :)

thank you for your review, I just follow your suggestion to remove psram_type of qspi. However, since I don't actually have this board yet. Could you mind waiting a bit longer for @ladyada to test on her prototypes.

**Comment by ladyada** [comment]
thanks for the update, just tested and i get 8MB psram so this can be merged :)

**Pull request merged**